### PR TITLE
`@drupalView` directive

### PR DESCRIPTION
**Pull request by Leksat**
## Package(s) involved

`amazeelabs/graphql_directives`

## Related Issue(s)

SLB-221

## How has this been tested?

Schema tests.

### CODE REVIEW

File: packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          'contextual_filters' => $contextualFilters,
+        ],
+        'rows' => $rows,
+        'total' => $total,
+        'filters' => $filters,
+      ];
+    });
+  }
+
+  function withCacheContext(FieldContext $resolveContext, callable $callback) {
```
Comment (pmelab) [inline packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php:125]: Sometimes I wonder if the GraphQL module should do that by default 😅

File: packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ *   description = @Translation("Executes a Drupal view."),
+ *   produces = @ContextDefinition("any",
+ *     label = @Translation("An array with view info and results"),
+ *     multiple = TRUE
+ *   ),
+ *   consumes = {
+ *     "id" = @ContextDefinition("string",
+ *       label = @Translation("View ID and display ID divided by a comma"),
+ *     ),
+ *     "args" = @ContextDefinition("string",
```
Comment (pmelab) [inline packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php:27]: We ended up with the same approach in https://github.com/AmazeeLabs/silverback-template/pull/147

Discussed with @chindris that deriving GraphQL types from Drupal config (e.g. views filters, webform inputs) could be interesting. Just as a reference.

**Review by pmelab — APPROVED** [review]
Can't come up with anything to complain about 😬

File: packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php (hunk 1 above)
Comment (Leksat) [reply packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php:125]: There's an ultimate solution: https://www.drupal.org/project/drupal/issues/2638686#comment-13818152

File: packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php (hunk 2 above)
Comment (Leksat) [reply packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php:27]: A little sad thing is that a tiny typo can break it unnoticeably. For example, if `tag` filter is multi-value, it should be passed as `tag[]=1`. If it's sent as `tag=1`, the value will be ignored.

Checking every possible filter with e2e tests does not sound like a great idea 😕

Comment (Leksat) [reply packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php:27]: Probably backend devs should provide TypeScript types for `args` 🤷

**Pull request merged**